src/benchmark/operators/sort_benchmark.cpp

```diff
@@ -3,20 +3,113 @@
 #include "benchmark/benchmark.h"
 
 #include "../micro_benchmark_basic_fixture.hpp"
+#include "SQLParser.h"
+#include "SQLParserResult.h"
+#include "expression/expression_functional.hpp"
+#include "hyrise.hpp"
+#include "logical_query_plan/lqp_translator.hpp"
+#include "operators/limit.hpp"
 #include "operators/sort.hpp"
 #include "operators/table_wrapper.hpp"
+#include "sql/sql_pipeline_builder.hpp"
+#include "sql/sql_translator.hpp"
+#include "synthetic_table_generator.hpp"
+
+#include "micro_benchmark_utils.hpp"
 
 namespace opossum {
 
-BENCHMARK_F(MicroBenchmarkBasicFixture, BM_Sort)(benchmark::State& state) {
-  _clear_cache();
+static std::shared_ptr<Table> generate_custom_table(const size_t row_count, const ChunkOffset chunk_size,
+                                                    const DataType data_type = DataType::Int,
+                                                    const std::optional<float> null_ratio = std::nullopt) {
+  const auto table_generator = std::make_shared<SyntheticTableGenerator>();
+
+  size_t num_columns = 2;
+
+  const int max_different_value = 10'000;
+  const std::vector<DataType> column_data_types = {num_columns, data_type};
+
+  std::vector<ColumnSpecification> column_specifications = std::vector(
+      num_columns, ColumnSpecification(ColumnDataDistribution::make_uniform_config(0.0, max_different_value), data_type,
+                                       {EncodingType::Unencoded}, std::nullopt, null_ratio));
+
+  return table_generator->generate_table(column_specifications, row_count, chunk_size, UseMvcc::Yes);
+}
+
+static void make_reference_table(std::shared_ptr<TableWrapper> table_wrapper) {
+  auto limit =
+      std::make_shared<Limit>(table_wrapper, expression_functional::to_expression(std::numeric_limits<int64_t>::max()));
+  limit->execute();
+  table_wrapper = std::make_shared<TableWrapper>(limit->get_output());
+  table_wrapper->execute();
+}
+
+static void BM_Sort(benchmark::State& state, const size_t row_count = 40'000, const ChunkOffset chunk_size = 2'000,
+                    const DataType data_type = DataType::Int, const std::optional<float> null_ratio = std::nullopt,
+                    const bool multi_column_sort = true, const bool use_reference_segment = false) {
```

Comment:
No need to change this, just a suggestion for the future: If all arguments have defaults, it might be cleaner to just pass a struct. That way, it is easier to just set the parameters that you want to vary without having to check the parameter order.